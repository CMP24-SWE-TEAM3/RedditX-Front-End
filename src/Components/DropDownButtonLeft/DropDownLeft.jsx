// imports
import React, {useState} from 'react';
import Dropdown from 'react-bootstrap/Dropdown';
import {DropDownLeftStyled} from "./DropDownLeft.styled";
import {AiFillHome} from 'react-icons/ai';
import DropDownMenu from "Components/DropDownMenu/DropDownMenu";
import CustomToggle from "Components/CustomToggle/CustomToggle";
import SideBar from "Components/SideBar/SideBar";
import {GrFormDown} from "react-icons/gr";


/**
 * Component that has all your accessibility and profile in drop-down list.
 *
 * @returns {React.Component}
 */

const DropDownLeft = () => {
    /**
     * @ function that controls to show and hide drop-down list
     * @param {boolean,boolean}
     */
    const [show, setShow] = useState(false);

    /**
     * @ function that handle toggleVisible of drop -down list and sidebar
     */
    const handleToggle = () => {
        if (!show && showSideBar)
            return;
        setShow(!show);
    };

    const [showSideBar, setShowSideBar] = useState(false);
    return (

        <DropDownLeftStyled>
            <Dropdown.Toggle as={CustomToggle} id="dropdown-home"
                             onClick={handleToggle}>
                <div className={'all-drop'}>
                    <div className={'home-ico'}>

                        <span>
                            <AiFillHome/>
                        </span>
                        <span>
                            <h6>Home</h6>
                        </span>
                    </div>
                    <SideBar showButton={show} showSideBar={showSideBar} setShowSideBar={setShowSideBar}/>
                </div>
                <span className={'down-form'}><GrFormDown/></span>
            </Dropdown.Toggle>
            {show && <DropDownMenu/>}
        </DropDownLeftStyled>

    );
};
export default DropDownLeft;



