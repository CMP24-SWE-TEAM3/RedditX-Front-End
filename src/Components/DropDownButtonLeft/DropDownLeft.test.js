import {shallow} from 'enzyme';

// import Components
import DropDownLeft from "./DropDownLeft";

describe("Drop Down list Component", () => {
    it("this is a test for Drop Down list Component", () => {
        expect(shallow(<DropDownLeft/>)).toMatchSnapshot();

    });
});