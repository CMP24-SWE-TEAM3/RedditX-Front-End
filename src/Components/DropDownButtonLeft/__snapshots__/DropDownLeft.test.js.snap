// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Down list Component this is a test for Drop Down list Component 1`] = `ShallowWrapper {}`;
